src/screens/Market/MarketDetail/index.tsx

```diff
@@ -185,19 +179,14 @@ export default function MarketDetail({
   */
 
   const navigateToSwap = useCallback(() => {
-    if (allAccounts && allAccounts.length === 1) {
-      navigation.navigate(NavigatorName.Swap, {
-        screen: ScreenName.Swap,
-      });
-    } else {
-      navigation.navigate(NavigatorName.AddAccounts, {
-        screen: ScreenName.AddAccountsSelectDevice,
-        params: {
-          currency: internalCurrency,
-        },
-      });
-    }
-  }, [navigation, internalCurrency, allAccounts]);
+    navigation.navigate(NavigatorName.Swap, {
+      screen: ScreenName.Swap,
+      params: {
+        defaultAccount:
+          allAccounts && allAccounts.length > 0 ? allAccounts[0] : undefined,
```

Comment:
ℹ️ just a suggestion, not a blocker

```suggestion
          allAccounts?.length > 0 ? allAccounts[0] : undefined,
```

Reply:
thanks i'm tired today 😅